gadgets/mri_core/NoiseAdjustGadget.h

```diff
@@ -10,20 +10,70 @@
 #include <complex>
 #include <ismrmrd/ismrmrd.h>
 #include <ismrmrd/xml.h>
+#include <nlohmann/json.hpp>
+#include "sfndam_serializable.h"
 
-namespace Gadgetron {
+using json = nlohmann::json;
 
-        struct NoiseCovariance {
-            ISMRMRD::IsmrmrdHeader header;
-            float noise_dwell_time_us;
-            hoNDArray<std::complex<float>> noise_covariance_matrix;
+namespace Gadgetron {
+    
+        struct NoiseCovariance : Gadgetron::Core::IO::SfndamSerializable<NoiseCovariance>{
+            void SerializeToSfndam(std::ostream& stream) const override
+            {
+                sfndam::sfndam<std::complex<float>> sf;
+                json j;
+                j["channels"] = channels_;
+                j["sample_count"] = sample_count_;
+                j["labels"] = labels_;
+                j["noise_dwell_time_us"] = noise_dwell_time_us_;
+                j["receiver_noise_bandwidth"] = receiver_noise_bandwidth_;
+                sf.meta = j.dump();
+                sf.array_dimensions = {static_cast<uint32_t>(channels_), static_cast<uint32_t>(channels_)};
+                sf.data.resize(matrix_.get_number_of_elements());
+                std::copy(matrix_.begin(), matrix_.end(), sf.data.begin());
+                sfndam::serialize(sf, stream);
+            }
+
+            static NoiseCovariance DeserializeFromSfnadm(std::istream& stream)
+            {
+                sfndam::sfndam<std::complex<float>> sf = sfndam::deserialize<std::complex<float>>(stream);
+                json j = json::parse(sf.meta);
+                auto labels = j["labels"].get<std::vector<std::string>>();
```

Comment:
The behavior is undefined if the key does not exist. https://json.nlohmann.me/api/basic_json/operator%5B%5D/
Better to use `at()` and maybe augment the error